ocs_ci/utility/ibmcloud.py

```diff
@@ -520,6 +522,268 @@ def delete_volume_id(self, volume):
+            timeout (int): Timeout for the command, defaults to 900 seconds.
+        """
+        logger.info("restarting nodes")
+
+        for node in nodes:
+            cmd = f"ibmcloud is instance-reboot {node.name} -f"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node restart command output: {out}")
+
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=timeout,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                )
+                sample.wait_for_func_status(result=constants.STATUS_RUNNING.lower())
+
+    def start_nodes(self, nodes):
+        """
+        Start the nodes on IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+        """
+        # logger.info(nodes)
+        for node in nodes:
+            # logger.info(node.get())
+            cmd = f"ibmcloud is instance-start {node.name}"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node start command output: {out}")
+
+    def stop_nodes(self, nodes, force=True, wait=True):
+        """
+        Stop the nodes on IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+            force (bool): True for VM ungraceful power off, False for graceful VM shutdown
+            wait (bool): Wait for the VMs to stop
+        """
+        for node in nodes:
+            cmd = f"ibmcloud is instance-stop {node.name} --force={force}"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node Stop command output: {out}")
+
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=300,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                )
+                sample.wait_for_func_status(result=constants.STATUS_STOPPED)
+
+    def restart_nodes_by_stop_and_start(
+        self, nodes, wait=True, force=True, timeout=300
+    ):
+        """
+        Restart nodes by stopping and starting VM in IBM Cloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True in case wait for status is needed, False otherwise
+            force (bool): True for force instance stop, False otherwise
+            timeout (int): Timeout for the command, defaults to 300 seconds.
+        """
+        logger.info(f"Stopping instances {list(nodes)}")
+        self.stop_nodes(nodes=nodes, force=force)
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=timeout,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                )
+                sample.wait_for_func_status(result=constants.STATUS_STOPPED)
```

Comment:
The `result` param should be a boolean value(See the function definition here: https://github.com/prsurve/ocs-ci/blob/6caf2667833e887f9df7392af3666067aa6493ee/ocs_ci/utility/utils.py#L1342). Maybe you can create a new method `is_node_running`, and use it here instead.

Reply:
Same thing for the other occurences.

Reply:
i have update the code